presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetEncoding.java

```diff
@@ -0,0 +1,230 @@
+import com.facebook.presto.hive.parquet.dictionary.ParquetDictionaryReader;
+import com.facebook.presto.hive.parquet.dictionary.ParquetDoubleDictionary;
+import com.facebook.presto.hive.parquet.dictionary.ParquetFloatDictionary;
+import com.facebook.presto.hive.parquet.dictionary.ParquetIntegerDictionary;
+import com.facebook.presto.hive.parquet.dictionary.ParquetLongDictionary;
+import parquet.bytes.BytesUtils;
+import parquet.column.ColumnDescriptor;
+import parquet.column.values.ValuesReader;
+import parquet.column.values.bitpacking.ByteBitPackingValuesReader;
+import parquet.column.values.boundedint.ZeroIntegerValuesReader;
+import parquet.column.values.delta.DeltaBinaryPackingValuesReader;
+import parquet.column.values.deltalengthbytearray.DeltaLengthByteArrayValuesReader;
+import parquet.column.values.deltastrings.DeltaByteArrayReader;
+import parquet.column.values.plain.BinaryPlainValuesReader;
+import parquet.column.values.plain.BooleanPlainValuesReader;
+import parquet.column.values.plain.FixedLenByteArrayPlainValuesReader;
+import parquet.column.values.plain.PlainValuesReader.DoublePlainValuesReader;
+import parquet.column.values.plain.PlainValuesReader.FloatPlainValuesReader;
+import parquet.column.values.plain.PlainValuesReader.IntegerPlainValuesReader;
+import parquet.column.values.plain.PlainValuesReader.LongPlainValuesReader;
+import parquet.column.values.rle.RunLengthBitPackingHybridValuesReader;
+import parquet.io.ParquetDecodingException;
+
+import java.io.IOException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static parquet.column.values.bitpacking.Packer.BIG_ENDIAN;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.BINARY;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.BOOLEAN;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY;
+import static parquet.schema.PrimitiveType.PrimitiveTypeName.INT32;
+
+public enum ParquetEncoding
+{
+    PLAIN
+            {
+                @Override
+                public ValuesReader getValuesReader(ColumnDescriptor descriptor, ParquetValuesType valuesType)
+                {
+                    switch (descriptor.getType()) {
+                        case BOOLEAN:
+                            return new BooleanPlainValuesReader();
+                        case BINARY:
+                            return new BinaryPlainValuesReader();
+                        case FLOAT:
+                            return new FloatPlainValuesReader();
+                        case DOUBLE:
+                            return new DoublePlainValuesReader();
+                        case INT32:
+                            return new IntegerPlainValuesReader();
+                        case INT64:
+                            return new LongPlainValuesReader();
+                        case INT96:
+                            return new FixedLenByteArrayPlainValuesReader(12);
```

Comment:
we can define a constant like `INT96_TYPE_LENGTH=12;` as `12` is also used below.